### PR TITLE
Sequelize plugin

### PR DESCRIPTION
**Pull request by nirsky**

### CODE REVIEW

File: packages/plugin-sequelize/src/sequelize.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+    private _createQueryPatch(original: Function) {
+        const thisPlugin = this;
+        return function (sql: any, option: any) {
+            if (thisPlugin._config?.ignoreOrphanedSpans && !thisPlugin?._tracer?.getCurrentSpan()) {
+                return original.apply(this, arguments);
+            }
+
+            const statement = sql?.query ? sql.query : sql;
+            const operation = option.type ?? (typeof statement === 'string' ? statement.split(' ')[0] : undefined);
```
Comment (habmic) [inline packages/plugin-sequelize/src/sequelize.ts:33]: Just to make sure I got it right, if the statement is:
select * from user, the operation would be 'select'?

File: packages/plugin-sequelize/src/sequelize.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            const attributes = {
+                [DatabaseAttribute.DB_SYSTEM]: sequelizeInstance.getDialect(),
+                [DatabaseAttribute.DB_USER]: config?.username,
+                [GeneralAttribute.NET_PEER_NAME]: config?.host,
+                [GeneralAttribute.NET_PEER_PORT]: config?.port ? String(config?.port) : undefined,
+                [GeneralAttribute.NET_TRANSPORT]: thisPlugin._getNetTransport(config?.protocol),
+                [DatabaseAttribute.DB_NAME]: config?.database,
+                [DatabaseAttribute.DB_OPERATION]: operation,
+                [DatabaseAttribute.DB_STATEMENT]: statement,
+                component: 'sequelize',
```
Comment (habmic) [inline packages/plugin-sequelize/src/sequelize.ts:46]: I think component is deprecated
@blumamir - WDYT?

File: packages/plugin-sequelize/src/sequelize.ts (hunk 1 above)
Comment (nirsky) [reply packages/plugin-sequelize/src/sequelize.ts:33]: not always but most of the times, depends on what the lib provides

File: packages/plugin-sequelize/package.json [hunk 3]
```diff
@@ -0,0 +1,76 @@
+{
+    "name": "opentelemetry-plugin-sequelize",
```
[the remark this answers is not in the document]
Comment (nirsky) [reply packages/plugin-sequelize/package.json:2]: we don't have it on the other libs.
If you wanna add it what to write? Aspecto?

File: packages/plugin-sequelize/src/sequelize.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+            const statement = sql?.query ? sql.query : sql;
+            const operation = option.type ?? (typeof statement === 'string' ? statement.split(' ')[0] : undefined);
+            const sequelizeInstance: sequelize.Sequelize = this;
+            const config = sequelizeInstance?.config;
+
+            const attributes = {
+                [DatabaseAttribute.DB_SYSTEM]: sequelizeInstance.getDialect(),
+                [DatabaseAttribute.DB_USER]: config?.username,
+                [GeneralAttribute.NET_PEER_NAME]: config?.host,
+                [GeneralAttribute.NET_PEER_PORT]: config?.port ? String(config?.port) : undefined,
```
Comment (blumamir) [inline packages/plugin-sequelize/src/sequelize.ts:41]: why not send the port as number?
I see that in HTTP it is sent as number

File: packages/plugin-sequelize/src/sequelize.ts (hunk 2 above)
Comment (blumamir) [inline packages/plugin-sequelize/src/sequelize.ts:46]: no need to send this attribute anymore.

File: packages/plugin-sequelize/src/sequelize.ts [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+                [GeneralAttribute.NET_TRANSPORT]: thisPlugin._getNetTransport(config?.protocol),
+                [DatabaseAttribute.DB_NAME]: config?.database,
+                [DatabaseAttribute.DB_OPERATION]: operation,
```
Comment (blumamir) [inline packages/plugin-sequelize/src/sequelize.ts:44]: according to the spec, this attribute should not be set

**Review by blumamir — APPROVED** [review]

File: packages/plugin-sequelize/src/sequelize.ts (hunk 2 above)
Comment (blumamir) [reply packages/plugin-sequelize/src/sequelize.ts:46]: indeed

File: packages/plugin-sequelize/package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (habmic) [reply packages/plugin-sequelize/package.json:2]: Yeah I think so

[the remark this answers is not in the document]
Comment (nirsky) [reply packages/plugin-sequelize/package.json:2]: done

File: packages/plugin-sequelize/src/sequelize.ts (hunk 2 above)
Comment (nirsky) [reply packages/plugin-sequelize/src/sequelize.ts:46]: talked to Amir - we will do a robust solution to replace "component" later on.

File: packages/plugin-sequelize/src/sequelize.ts (hunk 4 above)
Comment (nirsky) [reply packages/plugin-sequelize/src/sequelize.ts]: fixed

File: packages/plugin-sequelize/src/sequelize.ts (hunk 2 above)
Comment (nirsky) [reply packages/plugin-sequelize/src/sequelize.ts:46]: talked to Amir - we will do a robust solution to replace "component" later on.

File: packages/plugin-sequelize/src/sequelize.ts (hunk 5 above)
Comment (nirsky) [reply packages/plugin-sequelize/src/sequelize.ts:44]: will be kept for now

**Pull request merged**